### PR TITLE
Added check for URL and added clientside tests (fixes #89)

### LINKED ISSUE #89: Test mdns name is a URL
Currently we're returning any found mDNS service name without checking if it's a URL or not.

### PR DESCRIPTION
**Pull request by wilsonpage**

### CODE REVIEW

**Comment by wilsonpage** [comment]
@arcturus r?

File: test/scanner/metadata/index.test.js [hunk 1]
```diff
@@ -0,0 +1,95 @@
...
+    });
+
+    afterEach(function() {
+
+    });
+
+    describe('batches', function() {
+      beforeEach(function() {
+        this.calls = [
+          metadata('http://bbc.co.uk/news'),
```
Comment (samgiles) [inline test/scanner/metadata/index.test.js:36]: Do these tests depend on a response from these public URLs?

Comment (wilsonpage) [inline test/scanner/metadata/index.test.js:36]: No, `global.fetch` is stubbed. See above.

File: test/scanner/metadata/index.test.js [hunk 2]
```diff
@@ -0,0 +1,95 @@
...
+ * Dependencies
+ */
+
+const metadata = require('../../../lib/scanner/metadata');
+const assert = require('assert');
+const sinon = require('sinon');
+
+describe('metadata', () => {
+
+  describe('serverside', function() {
```
Comment (arcturus) [inline test/scanner/metadata/index.test.js:12]: nit: this describe word is not very descriptive.

**Comment by arcturus** [comment]
LGTM r+ shiiiiip it!

File: test/scanner/metadata/index.test.js (hunk 1 above)
Comment (samgiles) [inline test/scanner/metadata/index.test.js:36]: Nice! I completely missed that.

**Pull request merged**